### PR TITLE
Emukit structure doc

### PR DESCRIPTION
**Pull request by marpulli**
*Issue #, if available:* #12 

*Description of changes:* Doc describing of the different parts of the loop.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: doc/emukit_structure.rst [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+There are more specific loops for Bayesian optimization and experimental design that construct some of the component 
+parts for you.
+
+
+Model
+________
+All ``Emukit`` loops need a probabilistic model of the underlying system.
+Emukit does not provide functionality to build models as there are already many good modelling frameworks available in python.
+Instead, we provide a way of interfacing third part modelling libraries with Emukit. 
+We already provide a wrapper for using a mode created with ``GPy``.
```
Comment (mmahsereci) [inline doc/emukit_structure.rst]: mode -> model

File: doc/emukit_structure.rst [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+Different models and modelling frameworks will provide different functionality. 
+For instance a Gaussian process will usually have derivatives of the predictions available but random forests will not. 
+These different functionalities are represented by a set of interfaces which a model implements. 
+The basic interface that all models must implement is ``IModel``, which implements functionality to make predictions and
+update the model but a model may implement any number of other interfaces such as ``IDifferentiable`` which indicates a
+model has prediction derivatives available.
+
+Acquisition
+___________
+The acquisition function is a data collection policy that aims to maximise amount of information collected about some
```
Comment (mmahsereci) [inline doc/emukit_structure.rst]: the acquisition function is not the policy. The policy would be to choose the lowest point of the acquisition function. The acquisition is just a heuristic quantification of how valuable a future point might be.

Comment (marpulli) [reply doc/emukit_structure.rst]: Thanks, I was wondering how to phrase this. I'm going to steal the "heuristic quantification of how valuable a future point might be" bit.

File: doc/emukit_structure.rst [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+It can be either evaluated by the user or it can be passed into the loop and evaluated by Emukit.
+
+Model Updater
+_____________
+The ``ModelUpdater`` class updates the model with new training data after a new point is observed and optimizes any
+hyper-parameters of the model. 
+It can decide whether hyper-parameters need updating based on some internal logic.
+
+
+Stopping condition
```
Comment (apaleyes) [inline doc/emukit_structure.rst]: Capitalize "Condition"

File: doc/emukit_structure.rst [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+The acquisition is a heuristic quantification of how valuable collecting a future point might be. It is used by the candidate point calculator to decide which point(s) to collect next.
+
+Acquisition Optimizer
+_____________________
+The ``AcquisitionOptimizer`` optimizes the acquisition function to find the point at which the acquisition is a maximum.
+This will use the acquisition function gradients if they are available. 
+If gradients of the acquisition function are not available it will either estimate them numerically or use a gradient 
+free optimizer. 
+
+Candidate Point Calculator
```
Comment (apaleyes) [inline doc/emukit_structure.rst]: This should probably come before acquisition section, as it is referenced there

Comment (marpulli) [reply doc/emukit_structure.rst]: Done

File: doc/emukit_structure.rst (hunk 3 above)
Comment (marpulli) [reply doc/emukit_structure.rst]: Done

**Pull request merged**